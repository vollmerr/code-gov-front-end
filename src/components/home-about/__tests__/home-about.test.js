import React from 'react';

import HomeAbout from 'components/home-about';

import { shallow } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <HomeAbout /> component', () => {
  const props = {
    aboutItems: [{
        button: "test-button",
        description: "Test Description.",
        image: "test-image",
        link: "https://test-link.com",
        title: "test"
    }]
  }
  const store = mockStore(props);
  
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<HomeAbout store={store}/>);
  });
  it('Testing if component renders', () => {
    expect(wrapper.exists()).toEqual(true);
  });
  it('should render correct', () => {
    expect(wrapper).toMatchSnapshot();
  })
})