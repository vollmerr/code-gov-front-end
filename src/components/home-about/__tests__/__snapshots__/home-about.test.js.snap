// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HomeAbout /> component should render correct 1`] = `
<HomeAbout
  aboutItems={
    Array [
      Object {
        "button": "test-button",
        "description": "Test Description.",
        "image": "test-image",
        "link": "https://test-link.com",
        "title": "test",
      },
    ]
  }
  dispatch={[Function]}
  mission="test missions quote"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
